staging/src/k8s.io/kubectl/pkg/cmd/proxy/proxy.go

```diff
@@ -167,12 +167,12 @@ func (o *ProxyOptions) Complete(f cmdutil.Factory) error {
 			return err
 		}
 		if target.Path != "" && target.Path != "/" {
-			klog.Warning("Your kube context contains a server path " + target.Path + ", use --append-server-path to automatically append the path to each request")
+			klog.InfoS("Your kube context contains a server path which uses --append-server-path to automatically append the path to each request", "path", target.Path)
```

Comment:
Note that the semantic of the message changes. But let's first discuss the `Warning` -> `InfoS` conversion.

Most of the `klog.Warning` calls are hard to reach, but this one can be triggered with:
```
sudo sed -i -e 's;localhost:6444/;localhost:6444/somepath/;' /var/run/kubernetes/admin.kubeconfig
```

The output on master was:
```
$ go run ./cmd/kubectl proxy
W0124 12:36:30.594529  339860 proxy.go:170] Your kube context contains a server path /somepath/, use --append-server-path to automatically append the path to each request
Starting to serve on 127.0.0.1:8001
```

Now it is:
```
$ go run ./cmd/kubectl proxy
I0124 12:34:32.335900  337990 proxy.go:170] "Your kube context contains a server path which uses --append-server-path to automatically append the path to each request" path="/somepath/"
Starting to serve on 127.0.0.1:8001
```

I think the leading `W` for "warning" was so subtle that loosing it is not a big deal. Probably all such log message should explicitly have a `WARNING: Your kube context ...` prefix.

The more interesting question is whether user-facing output like this should be done via klog at all.

The "Starting to serve on" message is printed without klog. Should the same be done for everything that is printed unconditionally and logging only be used for output that is depending on `-v` >= 1?

/cc @soltysh @atiratree

Reply:
I spoke with @soltysh about this at KubeCon. The conclusion was that such output should not get emitted via klog. Instead, `fmt.Fprintf` should be used together without output streams `o.Out` or `o.ErrOut`, depending on the nature of the message. In this case here, `o.ErrOut` seems more suitable.

Reply:
Hi Patrick

Do you mean change
```
klog.Warning("Your kube context contains a server path " + target.Path + ", use --append-server-path to automatically append the path to each request")
```
to
```
fmt.Fprintf(o.ErrOut, "Your kube context contains a server path %s, use --append-server-path to automatically append the path to each request", target.Path)
```
?

If so, does it work for all "klog.Warning"?

Reply:
Yes. Basically everything that is klog.Info/Error/Warning instead of `klog.V().Info/Error/Warning` is part of the "normal" kubectl output and shouldn't use logging.